test/io/shapefile.jl

```diff
@@ -62,6 +62,10 @@
     # https://github.com/JuliaEarth/GeoIO.jl/issues/32
     @test GeoIO.load(joinpath(datadir, "issue32.shp")) isa AbstractGeoTable
     @test GeoIO.load(joinpath(datadir, "lines.shp")) isa AbstractGeoTable
+
+    # https://github.com/JuliaEarth/GeoIO.jl/issues/158
+    issue158table = @test_logs (:warn, r"rows dropped") GeoIO.load(joinpath(datadir, "issue158.shp"))
```

Comment:
Could we use `@test_warn` with the exact warning message in this test? That way we also test the number of rows discarded.